### PR TITLE
Reconcile location searches

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/ICDS-1200

##### PRODUCT DESCRIPTION
We use two different algorithms that populate locations-related dropdowns. This PR updates them all to use the better of the two, which supports exact search as described [here](https://confluence.dimagi.com/display/commcarepublic/Exact+Search+for+Locations).

The expanded Organization Structure page and the mobile worker filter (the search box in exports and elsewhere that lets you choose users or groups or locations) already used the good search. The following pages were using the other search and have been updated:

- Location assignment on edit mobile worker page
- Create new mobile worker
- Invite web user
- Filtered mobile worker download (feature flagged)
- Related locations (feature flagged)
- OpenMRS repeaters (feature flagged)

### CODE REVIEW

File: corehq/apps/reports/filters/users.py
Comment (mkangia) [inline corehq/apps/reports/filters/users.py]: (non blocking)
is there any advantage of keeping them on the class?
how about making a separate file called help texts that store such messages?
Not sure if that also would make them getting preloaded when you load the app and not each time when they are accessed via class.

File: corehq/apps/reports/filters/users.py
Comment (mkangia) [inline corehq/apps/reports/filters/users.py]: that's a weird one, where is the suffix used?

**Review by mkangia — APPROVED** [review]
Good to see a feature that was developed in eNikshay time being used everywhere. :)
Thanks for shipping this change @orangejenny

**Comment by mkangia** [comment]
Do we need to take approval from product for this?

**Review by kaapstorm — APPROVED** [review]

File: corehq/apps/reports/filters/users.py
Comment (orangejenny) [reply corehq/apps/reports/filters/users.py]: I don't know that there are enough such messages to make a separate file, going to leave this as-is for now.

File: corehq/apps/reports/filters/users.py
Comment (orangejenny) [reply corehq/apps/reports/filters/users.py]: In exports, where you have a single box that searches mobile users, web users, groups, and locations, it's useful to distinguish between them.

**Comment by orangejenny** [comment]
@dimagi/product fyi

**Review by esoergel — APPROVED** [review]

**Review by calellowitz — COMMENTED** [review]
Assuming my comment is no longer relevant, this looks great but will wait for confirmation before approval

File: corehq/apps/locations/views.py
Comment (orangejenny) [reply corehq/apps/locations/views.py]: I didn't like it either; updated to `display_types`.

File: corehq/apps/locations/views.py
Comment (orangejenny) [reply corehq/apps/locations/views.py]: fixed

File: corehq/apps/reports/filters/users.py
Comment (orangejenny) [reply corehq/apps/reports/filters/users.py]: I thought about this, but the other types are sub-types of user, like "Active Mobile Worker", "Deactivativated Mobile Worker", and "Web User", which seem useful even if a search is only for users.

**Review by esoergel — APPROVED** [review]

**Comment by orangejenny** [comment]
@esoergel @calellowitz Good call reminding me to test the whole API. It's the same for gathering initial results and for pagination, but not for querying. I added a few commits to fix this, and also to make the mobile workers page use the same js as the rest of the searches. This PR should now be ready to go.

File: corehq/apps/locations/static/locations/js/widgets.js
Comment (orangejenny) [reply corehq/apps/locations/static/locations/js/widgets.js]: ignoring

File: corehq/apps/locations/static/locations/js/widgets.js
Comment (orangejenny) [reply corehq/apps/locations/static/locations/js/widgets.js]: ignoring

**Comment by orangejenny** [comment]
The plugin ESLint warning is a bad sign, but I'm not going to worry about it unless/until it pops up on more PRs.

**Review by esoergel — APPROVED** [review]

**Comment by orangejenny** [comment]
@mkangia 

> Good to see a feature that was developed in eNikshay time being used everywhere. :)

Agreed!

**Comment by calellowitz** [comment]
Woohoo! Thanks for finishing that up

**Pull request merged**
